build.rs

```diff
@@ -10,7 +10,11 @@ fn main() {
 // This can be disabled with RUSTFLAGS="--cfg memchr_disable_auto_simd", but
 // this is generally only intended for testing.
 fn enable_simd_optimizations() {
-    if is_env_set("CARGO_CFG_MEMCHR_DISABLE_AUTO_SIMD") {
+    if is_env_set("CARGO_CFG_MEMCHR_DISABLE_AUTO_SIMD")
+        || !env::var("CARGO_CFG_TARGET_FEATURE")
+            .map(|features| features.contains("sse"))
+            .unwrap_or(false)
```

Comment:
Maybe move this conditional to a helper function with a sentence or two of docs explaining why we're gating SIMD on this?

Reply:
Done!